### PR TITLE
Cleanup scoped_ptr in src/common/dns_utils.cpp

### LINKED ISSUE #8732: Idea: use std::unique_ptr deleter to support custom destructor in custom smart pointer in dns_utils?
There is this line with a TODO:
https://github.com/monero-project/monero/blob/9489586add5116a48039616140bb7267b16ba7b1/src/common/dns_utils.cpp#L198

std::unique_ptr supports a custom deleter, but I am not sure if this fits all requirements.

### PR DESCRIPTION
**Pull request by 0xFFFC0000**
* No need for entire new class (Fix #8732).
	* Fix stdint.h header include in contrib/epee/include/net/http_base.h.

Fix #8732

### CODE REVIEW

**Comment by 0xFFFC0000** [comment]
Fail from `ci/gh-actions/depends / x86_64 Freebsd (pull_request) `  :

`curl: (22) The requested URL returned error: 404 Not Found`

**Comment by selsta** [comment]
@0xFFFC0000 the source code is hosted on getmonero which currently has an expired cert, should resolve itself later today

File: src/common/dns_utils.cpp [hunk 1]
```diff
@@ -327,10 +299,14 @@ std::vector<std::string> DNSResolver::get_record(const std::string& url, int rec
...
-
-  // destructor takes care of cleanup
-  ub_result_ptr result;
-
+  
+  ub_result *result;
+  // Make sure we are cleaning after result.
+  epee::misc_utils::auto_scope_leave_caller scope_exit_handler =
+    epee::misc_utils::create_scope_leave_handler([&](){
+    ub_resolve_free(result);
```
Comment (vtnerd) [inline src/common/dns_utils.cpp:307]: Should a `nullptr` check be added here?

**Review by vtnerd — APPROVED** [review]

File: src/common/dns_utils.cpp (hunk 1 above)
Comment (vtnerd) [reply src/common/dns_utils.cpp:307]: I guess it always worked before, so probably not.

Comment (0xFFFC0000) [reply src/common/dns_utils.cpp:307]: Good catch @vtnerd . 

To be honest, I thought about adding a simple nullptr check, but I didn't want to make any semantical changes to the code.

File: src/common/dns_utils.cpp [hunk 2]
```diff
@@ -327,10 +299,14 @@ std::vector<std::string> DNSResolver::get_record(const std::string& url, int rec
...
   dnssec_available = false;
   dnssec_valid = false;
-
-  // destructor takes care of cleanup
-  ub_result_ptr result;
-
+  
+  ub_result *result;
+  // Make sure we are cleaning after result.
+  epee::misc_utils::auto_scope_leave_caller scope_exit_handler =
```
Comment (jeffro256) [inline src/common/dns_utils.cpp:305]: Why not combine these into a `std::unique_ptr` with a custom deleter?

Comment (0xFFFC0000) [reply src/common/dns_utils.cpp:305]: No easy way of handling `**` argument via `unique_ptr` [1]. The `result` is used as parameter [2].


1. https://stackoverflow.com/a/24543176
2. https://github.com/monero-project/monero/pull/9090/files/430b102aef0151820a27eafe232afa40ba8e3d8f#diff-ed01ce7362fc3c2d5c7adc0a0be9742bbc6c7558825914f6a2d288b1e0f552fcR313

File: src/common/dns_utils.cpp (hunk 1 above)
Comment (0xFFFC0000) [reply src/common/dns_utils.cpp:307]: I am resolving this conversation. Please feel free to reopen this comment.

**Review by jeffro256 — APPROVED** [review]

**Pull request merged**